thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyTimeBasedSummarizer.java

```diff
@@ -71,16 +74,18 @@ public int compare(RawAnomalyResultDTO o1, RawAnomalyResultDTO o2) {
         populateMergedResult(mergedAnomaly, currentResult);
       } else {
         // compare current with merged and decide whether to merge the current result or create a new one
-        if (applySequentialGapBasedSplit
-            && (currentResult.getStartTime() - mergedAnomaly.getEndTime()) > sequentialAllowedGap) {
+        MergedAnomalyResultDTO currMergedAnomaly = new MergedAnomalyResultDTO();
+        populateMergedResult(currMergedAnomaly, currentResult);
+        if ((applySequentialGapBasedSplit
```

Comment:
This condition may need some comment to explanation: you defined isMergeable here, but (Not) isMergeable and other two conditions (which defines isOverlap) together defines (shouldBeSeparated).